src/main/java/frc/robot/vision/limelights/LimelightFilters.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.vision.limelights;
+
+import edu.wpi.first.math.geometry.*;
+import frc.robot.constants.Field;
+
+public class LimelightFilters {
+
+	protected static boolean
+		keepLimelightData(LimelightRawData limelightRawData, Pose2d currentEstimatedPose, LimelightFiltersTolerances tolerances) {
+		return LimelightFilters.isAprilTagInProperHeight(limelightRawData, tolerances.aprilTagHeightToleranceMeters())
+			&& LimelightFilters.isLimelightOutputInTolerance(
+				limelightRawData,
+				currentEstimatedPose,
+				tolerances.positionTolerance(),
+				tolerances.rotationTolerance()
+			)
+			&& LimelightFilters.isRollInTolerance(limelightRawData, tolerances.rollTolerance())
+			&& LimelightFilters.isPitchInTolerance(limelightRawData, tolerances.pitchTolerance())
+			&& LimelightFilters.isRobotOnGround(limelightRawData, tolerances.robotToGroundToleranceMeters());
+	}
+
+	protected static boolean isLimelightOutputInTolerance(
+		LimelightRawData limelightRawData,
+		Pose2d estimatedPose,
+		double positionTolerance,
+		double rotationTolerance
```

Comment:
```suggestion
		Rotation2d rotationTolerance
```

Reply:
but thats not a Rotation2d, thats a normal, a value between -1 and 1